Tests/test_lib_image.py

```diff
@@ -30,3 +60,69 @@ def test_setmode():
         im.im.setmode("L")
     with pytest.raises(ValueError):
         im.im.setmode("RGBABCDE")
+
+
+@pytest.mark.parametrize("mode", mode_names_not_bgr)
+def test_equal(mode):
+    num_img_bytes = len(Image.new(mode, (2, 2)).tobytes())
+    # alternatively, random.randbytes() in Python 3.9
+    data = secrets.token_bytes(num_img_bytes)
+    img_a = Image.frombytes(mode, (2, 2), data)
+    img_b = Image.frombytes(mode, (2, 2), data)
+    assert img_a.tobytes() == img_b.tobytes()
+    assert img_a.im == img_b.im
+
+
+# With mode "1" different bytes can map to the same value,
+# so we have to be more specific with the values we use.
+def test_not_equal_mode_1():
+    data_a = data_b = bytes(secrets.choice(b"\x00\xff") for i in range(4))
+    while data_a == data_b:
+        data_b = bytes(secrets.choice(b"\x00\xff") for i in range(4))
+    # We need to use rawmode "1;8" so that each full byte is interpreted as a value
+    # instead of the bits in the bytes being interpreted as values.
+    img_a = Image.frombytes("1", (2, 2), data_a, "raw", "1;8")
+    img_b = Image.frombytes("1", (2, 2), data_b, "raw", "1;8")
+    assert img_a.tobytes() != img_b.tobytes()
+    assert img_a.im != img_b.im
+
+
+@pytest.mark.parametrize("mode", [mode for mode in mode_names_not_bgr if mode != "1"])
+def test_not_equal(mode):
+    num_img_bytes = len(Image.new(mode, (2, 2)).tobytes())
+    # alternatively, random.randbytes() in Python 3.9
+    data_a = data_b = secrets.token_bytes(num_img_bytes)
+    while data_a == data_b:
+        data_b = secrets.token_bytes(num_img_bytes)
+    img_a = Image.frombytes(mode, (2, 2), data_a)
+    img_b = Image.frombytes(mode, (2, 2), data_b)
+    assert img_a.tobytes() != img_b.tobytes()
+    assert img_a.im != img_b.im
+
+
+@pytest.mark.parametrize(
+    ("mode", "rawmode"),
+    (("RGB", "RGBX"), ("YCbCr", "YCbCrX"), ("HSV", None), ("LAB", None)),
```

Comment:
It's more semantically.
```suggestion
    [("RGB", "RGBX"), ("YCbCr", "YCbCrX"), ("HSV", None), ("LAB", None)],
```

Reply:
A previous decision was to use tuples. https://github.com/python-pillow/Pillow/issues/6525#issuecomment-1272405965

Reply:
I've provided a detailed answer there. Hope @radarhere will agree with me

Reply:
As an example you can look at the [Django settings documentation](https://docs.djangoproject.com/en/4.2/ref/settings/). Settings are almost always immutable, but both lists and tuples are used for different things. Tuples are used for [ADMINS](https://docs.djangoproject.com/en/4.2/ref/settings/#admins) items (first value is a name, second is an email), [LANGUAGES](https://docs.djangoproject.com/en/4.2/ref/settings/#languages) items, [SECURE_PROXY_SSL_HEADER](https://docs.djangoproject.com/en/4.2/ref/settings/#secure-proxy-ssl-header) (first value is a name, second is a value of header). While lists are used for all sorts of… well… lists.

Reply:
I'm not overly concerned about which is used, and I don't think performance considerations are exactly a priority for tests, so I'm happy to go with @homm's preference here.